elf/module.go

```diff
@@ -504,20 +503,22 @@ func (b *Module) closeSocketFilters() error {
 	return nil
 }
 
-func unpinMap(m *Map) error {
-	if m.m.def.pinning == 0 {
-		return nil
+func unpinMap(m *Map, pinPath string) error {
+	mapPath, err := getMapPath(&m.m.def, m.Name, pinPath)
+	if err != nil {
+		return err
 	}
-	namespace := C.GoString(&m.m.def.namespace[0])
-	mapPath := filepath.Join(BPFFSPath, namespace, BPFDirGlobals, m.Name)
 	return syscall.Unlink(mapPath)
 }
 
 func (b *Module) closeMaps(options map[string]CloseOptions) error {
 	for _, m := range b.maps {
 		doUnpin := options[fmt.Sprintf("maps/%s", m.Name)].Unpin
 		if m.m.def.pinning > 0 && doUnpin {
-			unpinMap(m)
+			pinPath := options[fmt.Sprintf("maps/%s", m.Name)].PinPath
+			if err := unpinMap(m, pinPath); err != nil {
+				return fmt.Errorf("error unpinning path %q: %v", pinPath, err)
```

Comment:
This would give a misleading error message in case unpinning of a `PIN_GLOBAL_NS` map fails (w/o a `PinPath`).